scripts/west_commands/sign.py

```diff
@@ -237,6 +237,264 @@ def sign(self, command, build_dir, build_conf, formats):
+                        " built for code-partition that has not been placed"
+                        " on internal flash")
+
+            return partition
+
+    def __init__(self, quiet=False):
+        self.quiet = quiet
+
+
+    @staticmethod
+    def mcuboot_check_alignment(alignment):
+        '''Minimal MCUboot write alignment is 8 bytes, maximum is 32
+
+        :parameter alignment: flash alignment
+        :type alignment: int
+        :returns: alignment, corrected if required
+        :rtype: int
+        '''
+
+        if alignment < 8:
+            return 8
+
+        if alignment > 32:
+            log.die("MCUboot does not work with flash with write alignment > 32")
+
+        return alignment
+
+    @staticmethod
+    def mcuboot_swap_validate(mcbdtconf, bin_size, quiet=False):
+        '''Check whether DT configuration of partitions is valid for MCUboot swap-move algorithm.
+
+        If configuration is valid, returns parameters needed by imgtool sign command.
+
+        :param mcbdtconf: MCUboot configuration reader
+        :type: MCUbootDTConfig
+        :param bin_size: size of application binary; this is size application will take
+                         on flash.
+        :type bin_size: int
+        :param quiet: suppress extra diagnostic information
+        :type quiet: Bool
+        :returns: (flash alignment, application image offset on flash, image size)
+        :rtype: (int, int, int)
+        '''
+
+        code = mcbdtconf.get_code_partition()
+
+        if 'slot0_partition' not in code.labels:
+            log.die("west sign, for MCUboot in swap mode, expects DT chosen"
+                    " zephyr,code-partition to point to node labeled"
+                    " slot0_partition. Currently selected node is labeled"
+                    " " + str(code.labels))
+
+        if 'slot1_partition' in code.labels:
+            log.die("chosen zephyr,code-partition may not have slot0_partition"
+                    " and slot1_partition at the same time")
+
+        (write_alignment, _) = code.flash_parameters()
+
+        if write_alignment == 0:
+            log.die('expected non-zero flash alignment, but got'
+                    f' DT flash device write-block-size {write_alignment} for {code}')
+
+        write_alignment = __class__.mcuboot_check_alignment(write_alignment)
+
+        cp_addr, cp_size = code.get_parameters()
+
+        other = mcbdtconf.get_partition('slot1_partition')
+
+        if other:
+            _, op_size = other.get_parameters()
+            # Always go with smaller partition size, as swap will not be possible
+            if op_size != cp_size:
+                log.wrn("slot0_partition and slot1_partition differ in size:"
+                        f" ${cp_size} vs ${op_size}. Smaller size will be used.")
+                if op_size < cp_size:
+                    cp_size = op_size
+        else:
+            log.wrn('slot1_partition not found, assuming single slot configuration')
```

Comment:
I think this might become a bit verbose to see this every time if the user is using single slot mode, debug level log might be better suited